### PR TITLE
eww: 0.6.0 -> 0.6.0-unstable-2024-04-26

### LINKED ISSUE #308473: eww 0.6.0 string truncation problems?
### Describe the bug
After upgrading to eww 0.6.0 from unstable I've got problems showing some strings in some of my widgets, one that shows the hour and another the minutes using two digits each one, i.e. : "09 : 10".

When using eww 0.6.0 the string shown is like this: "... : ..."

In eww's github repo I can see that there is a commit after version 0.6.0:

https://github.com/elkowar/eww/commit/2c8811512460ce6cc75e021d8d081813647699dc

And cloning eww repo with that commit, compiling it and using a local eww executable the problem explained before disappears for me.

There's also another problem, asking eww-0.6.0 for version number shows it's @version 0.5.0:

```console
[user@system:~]$  /nix/store/ws61aiclafaga48kb59bgpmh08q6ya5p-eww-0.6.0/bin/eww --version
eww 0.5.0
```

### Steps To Reproduce
1. install unstable eww 0.6.0
2. Use some string truncation (I think this is the problem)
3. I see "..." where a two char string with the hour should appear.

### Expected behavior
eww should work as expected.

### Additional context
Cloning eww repo with latest commit after v0.6.0 was published solves the problem for me.

### Metadata 
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 6.8.8, NixOS, 23.11 (Tapir), 23.11.6759.0638fe2715d9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.2`
 - channels(root): `"nixos-23.11, nixpkgs-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

### PR DESCRIPTION
**Pull request by JohnRTitor**
## Description of changes
This PR merely backports the upstream patch (https://github.com/elkowar/eww/commit/2c8811512460ce6cc75e021d8d081813647699dc) to fix the problem in current release.
Also reformats the file using `nixfmt-rfc-style`.
Fixes https://github.com/NixOS/nixpkgs/issues/308473

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ew/eww/package.nix [hunk 1]
```diff
@@ -32,18 +46,27 @@ rustPlatform.buildRustPackage rec {
...
+  ];
 
   cargoTestFlags = cargoBuildFlags;
 
   # requires unstable rust features
   RUSTC_BOOTSTRAP = 1;
 
   meta = {
-    description = "ElKowars wacky widgets";
+    description = "Elkowars Wacky Widgets, a widget system made in Rust to allow you create widgets for any WM.";
```
Comment (eclairevoyant) [inline pkgs/by-name/ew/eww/package.nix:60]: This can go in longDescription but not description, this is a bit long and doesn't follow guidelines (shouldn't end in period). Alternatively you can rephrase this to be shorter

File: pkgs/by-name/ew/eww/package.nix [hunk 2]
```diff
@@ -17,13 +19,25 @@ rustPlatform.buildRustPackage rec {
   src = fetchFromGitHub {
     owner = "elkowar";
     repo = "eww";
-    rev = "refs/tags/v${version}";
+    rev = "v${version}";
```
Comment (eclairevoyant) [inline pkgs/by-name/ew/eww/package.nix:22]: remove this change

File: pkgs/by-name/ew/eww/package.nix [hunk 3]
```diff
@@ -17,13 +19,25 @@ rustPlatform.buildRustPackage rec {
...
     hash = "sha256-rzDnplFJNiHe+XbxbhZMEhPJMiJsmdVqtZxlxhzzpTk=";
   };
 
+  patches = [
+    # FIXME: remove with the next update
```
Comment (eclairevoyant) [inline pkgs/by-name/ew/eww/package.nix:27]: Would be best to also explain the patch's purpose, or at least link to the issue

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the PR. Some changes below:

File: pkgs/by-name/ew/eww/package.nix (hunk 2 above)
Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix:22]: Can you consider explaining why? fetchFromGitHub can find the tag just fine if I leave it like this.

Comment (eclairevoyant) [reply pkgs/by-name/ew/eww/package.nix:22]: There's absolutely no reason for this change. And it's better to be explicit especially if branches get created in the future as well as for the sake of update scripts.

File: pkgs/by-name/ew/eww/package.nix (hunk 3 above)
Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix]: Done.

File: pkgs/by-name/ew/eww/package.nix (hunk 2 above)
Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix]: Reverted, thanks.

File: pkgs/by-name/ew/eww/package.nix (hunk 1 above)
Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix]: Rephrased the description to be shorter
Moved the long stuff to longDescription, does this look fine now?

File: pkgs/by-name/ew/eww/package.nix [hunk 4]
```diff
@@ -21,9 +23,22 @@ rustPlatform.buildRustPackage rec {
...
 
+  patches = [
+    # FIXME: remove with the next release
+    # https://github.com/elkowar/eww/pull/1084 - patch to fix string truncation in eww config
+    (fetchpatch {
+      name = "fix-eww-string-truncation.patch";
+      url = "https://github.com/elkowar/eww/commit/2c8811512460ce6cc75e021d8d081813647699dc.patch";
+      hash = "sha256-oYrpuTI8KbMtboOdc8xw3UrAEkLu5xnphgey7koKmrc=";
+    })
+  ];
```
Comment (w-lfchen) [inline pkgs/by-name/ew/eww/package.nix:34]: Is there a reason for not just changing the source ref to the commit you're fetching here?

File: pkgs/by-name/ew/eww/package.nix [hunk 5]
```diff
@@ -32,18 +47,34 @@ rustPlatform.buildRustPackage rec {
...
   RUSTC_BOOTSTRAP = 1;
 
   meta = {
-    description = "ElKowars wacky widgets";
+    description = "A widget system made in Rust to create widgets for any WM";
+    longDescription = ''
+      Eww (ElKowar's Wacky Widgets) is a widget system made in Rust, which lets
+      you create your own widgets similarly to how you can in AwesomeWM.
+      The key difference: It is independent of your window manager!
+      It can be configured in yuck and themed using CSS, very easy
```
Comment (w-lfchen) [inline pkgs/by-name/ew/eww/package.nix:66]: I'd change this to "... using CSS, **is** very easy ..."

File: pkgs/by-name/ew/eww/package.nix [hunk 6]
```diff
@@ -32,18 +47,34 @@ rustPlatform.buildRustPackage rec {
...
   cargoTestFlags = cargoBuildFlags;
 
   # requires unstable rust features
   RUSTC_BOOTSTRAP = 1;
 
   meta = {
-    description = "ElKowars wacky widgets";
+    description = "A widget system made in Rust to create widgets for any WM";
+    longDescription = ''
+      Eww (ElKowar's Wacky Widgets) is a widget system made in Rust, which lets
```
Comment (w-lfchen) [inline pkgs/by-name/ew/eww/package.nix:63]: This comma feels wrong to me but I am no English native speaker so I could be mistaken

File: pkgs/by-name/ew/eww/package.nix (hunk 4 above)
Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix:34]: I thought the same, since it also [incorporates](https://github.com/elkowar/eww/compare/v0.6.0...2c8811512460ce6cc75e021d8d081813647699dc) Cargo.toml and other changes I was hesitant.
Second opinion, @eclairevoyant?

File: pkgs/by-name/ew/eww/package.nix (hunk 6 above)
Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix:63]: Well, I've always used it that way :)
better to have another opinion

File: pkgs/by-name/ew/eww/package.nix (hunk 4 above)
Comment (w-lfchen) [reply pkgs/by-name/ew/eww/package.nix:34]: I'd be really surprised if those changes cause issues

Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix:34]: Yep, thought the same :) looked at the changes, pretty harmless bugfixes and deps update

Comment (eclairevoyant) [reply pkgs/by-name/ew/eww/package.nix:34]: Either is fine, though if you're changing the commit, you'll also have to change the version to the unstable format.

File: pkgs/by-name/ew/eww/package.nix (hunk 6 above)
Comment (eclairevoyant) [reply pkgs/by-name/ew/eww/package.nix:63]: I agree with both of w-lfchen's suggestions about the description.

Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix]: Done.

File: pkgs/by-name/ew/eww/package.nix (hunk 5 above)
Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix]: Done.

File: pkgs/by-name/ew/eww/package.nix (hunk 4 above)
Comment (JohnRTitor) [reply pkgs/by-name/ew/eww/package.nix]: Using the version `0.6.0-unstable-2024-04-26`.

**Review by w-lfchen — APPROVED** [review]
LGTM

**Review by eclairevoyant — APPROVED** [review]
